### ISSUE TITLE
[Request] show and change print settings before print

### ISSUE THREAD

**Issue by korneelb** [OctoPrint/OctoPrint#653]
Hi;

as someone who uses the Octoprint solution in combination with different materials, but the same gcodes, this is something i think would be very handy;

when you load the gcode into Octoprint for printing it shows you all details such as filament length used etc. it would be nice to have a little drop down box there to select one of your temperature profiles instead of using the one out of the gcode. that way you can use the same gcode for all filaments.

**Comment by markwal** [comment]
Brainstorming:

I'm thinking it would be nifty to make a plugin that allows managing a library of materials.  Each material would have info like nozzle temp, bed temp, retraction distance, flow rate, cooling fan setting.  Then in order to have it have some effect, define some templates variables for the gcode and expand the templates as the gcode comes through.

The tricky bit will be convincing the slicer to put the template expansion in there.  Most of these things can come from the slicers gcode scripts, but since they already do template expansion of their own, we'd need to pick something that would pass through the slicer.  Even niftier would be a way such that the gcode is actually valid with defaults and the variable expansion is used as override.

Then, we'd want a UI panel where you set the currently loaded filament set for each nozzle.  Perhaps just added controls to the Control tab.

**Comment by ManuGithubSteam** [comment]
Yes i would need this also. I have to control the flow of filament for different types of filament but for now there is no profile setting. Maybe we could use the predefined temps of the bed and hotend to also include flow rate as it will mostly differ also. 

Unfortunately i don't know how to use send POST commands to the api otherwise i may be able to make a super simple plugin....
